project/src/main/webapp/script.js

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.import java.io.IOException;
+
+/**
+ * Fetches recommended places from the 'query' servlet, and switches from the query form to the
+ * results elements in order to display them to the user.
+ */
+function fetchFromQuery() {
+  document.getElementById('query-form').style.display = 'none';
+  document.getElementById('results').style.display = 'block';
+  const placesDiv = document.getElementById('place');
+  fetch('/query').then(response => response.json()).then((places) => {
+    places.forEach((singlePlace) => {
+      placesDiv.appendChild(createPlaceElement(singlePlace));
+    });
+  });
+}
+
+/**
+ * Creates a place element.
+ */
+function createPlaceElement(place) {
+  const placeElement = document.createElement('div');
+  placeElement.class = 'place-container';
+
+  // Add name
+  const name = document.createElement('li');
+  name.innerText = place.name;
+  placeElement.appendChild(name);
+  placeElement.appendChild(document.createElement('br'));
+
+  // Add link to website
```

Comment:
I found cases where placeDestails.website is null on Google Places. For now I handled it by setting an empty string on the Place.websiteURL field. So in this case the user will see the 'Restaurant's website' title without it leading to a link.
We need to make sure that we either filter out places like that so only places that have all fields get here, or not display 'Restaurant's website' if it's url is "".
What do you think we should do?

Reply:
That's a great question. I think we'll have to make a decision about the first option, because it depends on other things, for example if we have a link to Google Maps that might be good enough. So for now I added a case that allows to display places without a website, and I think later on we'll decide about the filtering.